specs/xaml-resource-references-tracing-spec.md

```diff
@@ -0,0 +1,189 @@
+
+A XAML resource is an item in a 
+[ResourceDictionary](https://docs.microsoft.com/windows/windows-app-sdk/api/winrt/Microsoft.UI.Xaml.ResourceDictionary)
+that can be referenced in markup with a `{StaticResource}` or `{ThemeResourcd}` reference.
+For example:
+
+```xml
+<StackPanel >
+    <StackPanel.Resources>
+        <SolidColorBrush x:Key="myBrush" Color="Red"/>
+    </StackPanel.Resources>
+
+    <Rectangle Fill="{StaticResource myBrush}"/>
+```
+
+`ResourceDictionary`s can be defined in multiple places, so the resource reference is resolved
+as a search.
+
+Failure to resolve a XAML resource reference is one of 
+the most common causes of app crashes. Such failures manifest in the _native code_ debug output with the 
+message "Cannot find a Resource with the Name/Key *foo*", and generally fall into one of two 
+buckets:
+
+1. The referenced resource legitimately does not exist, e.g. the referenced key is misspelled or 
+the intended matching resource has not been added to the app.
+2. The referenced resource *does* exist in the app, but it is not reachable from the reference 
+at run-time.
+
+Unfortunately, the stowed exception message is the *only* information provided about the error, 
+which makes it difficult or, more often, outright impossible to debug.
+
+This spec describes a new 
+API on the `DebugSettings` class that developers can use to access more detailed information about 
+the resource reference lookup failure.
+The new APIs here are patterned after the existing
+[DebugSettings.BindingFailed](https://docs.microsoft.com/windows/windows-app-sdk/api/winrt/Microsoft.UI.Xaml.DebugSettings.BindingFailed)
+and
+[DebugSettings.IsBindingTracingEnabled](https://docs.microsoft.com/windows/windows-app-sdk/api/winrt/Microsoft.UI.Xaml.DebugSettings.IsBindingTracingEnabled)
+APIs.
+
+
+# API Pages
+
+_(Updates to the 
+[DebugSettings](https://docs.microsoft.com/windows/windows-app-sdk/api/winrt/Microsoft.UI.Xaml.DebugSettings)
+class)_
+
+## DebugSettings.XamlResourceReferenceFailed Event
+
+Occurs when a
+[XAML resource reference](https://learn.microsoft.com/en-us/windows/apps/design/style/xaml-resource-dictionary)
+cannot be resolved.
+
+_Spec note: This API is similar to the `DebugSettings.BindingFailed` event. Like XAML resource references, 
+Bindings cannot be modeled solely through static evaluation, e.g. at compile-time, and so a run-time 
+event when a failure occurs is the best approach for providing developers with a means of determining the 
+root cause._
+
+_Spec note: is there a version of the 'ResourceDictionary and XAML resource references' article that 
+links to the Windows App SDK documentation for relevant APIs?_
+
+```c#
+public event TypedEventHandler<DebugSettings,XamlResourceReferenceFailedEventArgs> 
+XamlResourceReferenceFailed
+
+```
+
+### Remarks
+
+`IsXamlResourceReferenceTracingEnabled` must be `true` in order for this event to be raised.
+
+Error information is also logged to the native debug output,
```

Comment:
Later text suggests that I have to set IsXamlResourceReferenceTracingEnabled = true to get native debug output.

> Error information is also logged to the native debug output if `IsXamlResourceReferenceTracingEnabled` is `true`,